cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -216,6 +224,9 @@ public void launch(RuntimeJobInfo runtimeJobInfo) throws Exception {
     // In dataproc bucket, the shared folder for artifacts will be <bucket>/cdap-job/cached-artifacts.
     // All instances of CacheableLocalFile will be copied to the shared folder if they do not exist.
     String cacheRootPath = getPath(DataprocUtils.CDAP_GCS_ROOT, DataprocUtils.CDAP_CACHED_ARTIFACTS);
+    // cdap_version is of the form "major.minor.fix-buildTime"/"major.minor.fix-SNAPSHOT-buildTime"
+    String[] cdapVersion = cdapVersionInfo.toString().split("-");
```

Comment:
We actually have build time as a property. See the `ProjectInfo`. I think we can expose more information from the `VersionInfo` class in the SPI to simplify this. Parsing a string is never a good idea as the tostring contract can change.

Reply:
ack